jOOQ/src/main/java/org/jooq/impl/Tools.java

```diff
@@ -1421,11 +1421,12 @@ else if (value instanceof QueryPart)
      *         values created from the argument objects.
      */
     static final <T> List<Field<T>> fields(T[] values) {
-        List<Field<T>> result = new ArrayList<Field<T>>();
+        int length = values == null ? 0 : values.length;
```

Comment:
This null check will lead to potential null pointer access warnings further down. I'll change this into:

```java
if (values == null)
    return new ArrayList<Field<T>>();
```